tests/platform_tests/api/test_module.py

```diff
@@ -84,10 +84,19 @@ def get_skip_mod_list(self, duthosts, enum_rand_one_per_hwsku_hostname):
         duthost = duthosts[enum_rand_one_per_hwsku_hostname]
         self.skip_mod_list = get_skip_mod_list(duthost)
 
+    def skip_absent_module(self, module_num, platform_api_conn):
+        name = module.get_name(platform_api_conn, module_num)
+        if name in self.skip_mod_list:
+            logger.info("Skipping module {} since it is part of skip_mod_list".format(name))
+            return True
+        return False
+
     def test_get_name(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, platform_api_conn):
 
         for i in range(self.num_modules):
             name = module.get_name(platform_api_conn, i)
+            if self.skip_absent_module(i,platform_api_conn):
```

Comment:
I think you should move this to immediately after you enter the for loop and only if it passed then execute the original code to get name and others...